addon/Simulator.jsm

```diff
@@ -0,0 +1,142 @@
+  return logger;
+}
+
+/**
+ * Runs the simulation and writes the data into case_reports.csv.
+ * For running the simulation, the true values from case_true_values.csv
+ * are read, and RAPPOR is executed.
+ * The expected format of case_true_values.csv is {client, cohort, value}.
+ * It contains the true values used for the simulation, the client that submits
+ * the value and the cohort this client belongs to.
+ * The format of case_reports.csv is {client, cohort, bloom, prr, irr}. This values
+ * are needed to run the analysis.
+ * @param {string} studyName - Name of the study.
+ * @param {string} rapporPath - Path where the RAPPOR simulator lives.
+ * @param {object} params - Object containing the algorithm parameters.\
+ * @param {constant} method  - Method used to encode. Supported: nsICryptoHash.MD5 and nsICryptoHash.SHA256
+ * @param {string} instance - Simulation instance.
+ */
+function runRapporSimulation(studyName, rapporPath, params, method, instance) {
+  let data = read(new FileUtils.File(rapporPath + "_tmp/python/" + instance + "/1/case_true_values.csv"));
+  let caseReportsFile = new FileUtils.File(rapporPath + "_tmp/python/" + instance + "/1/case_reports.csv");
+  write(caseReportsFile, "client, cohort, bloom, prr, irr\n");
+  // Iterate over each line of the file getting the client, cohort and value.
+  // Each line represents a candidate sample that will be used to test the rappor implementation.
+  for (let i = 1; i < data.length; i++) {
+    let line = data[i].split(",");
+    let report = TelemetryRappor.createReport(studyName, line[2], params, method, line[1]);
+    // The expected format is {client, cohort, bloom, prr, irr}.
+    write(caseReportsFile, line[0] + ","+ line[1] + "," + convertToBin(report.internalBloom) +
+          "," + convertToBin(report.internalPrr) + "," + convertToBin(report.report) + "\n");
+  }
+}
+
+/**
+ * Read the cohort and the true value from a file. {client, cohort, value}.
+ * @param {nsFile} file - file containing the true values.
+ */
+function read(file) {
+  // open an input stream from file
+  var istream = Cc["@mozilla.org/network/file-input-stream;1"].createInstance(Ci.nsIFileInputStream);
+  istream.init(file, 0x01, 0o444, 0);
+  istream.QueryInterface(Ci.nsILineInputStream);
+  // read lines into array
+  var line = {}, lines = [], hasmore;
+  do {
+    hasmore = istream.readLine(line);
+    lines.push(line.value);
+  } while(hasmore);
+
+  istream.close();
+
+  return lines;
+}
+
+/**
+ * Write in a CSV {client, cohort, bloom, prr, irr}.
+ * @param {nsIFile} file - File to write in.
+ * @param {string} data - String containing the client, cohort, bloom, prr and irr.
+ */
+function write(file, data) {
+  // file is nsIFile, data is a string
+  var foStream = Cc["@mozilla.org/network/file-output-stream;1"].createInstance(Ci.nsIFileOutputStream);
+
+  foStream.init(file, 0x02 | 0x08 | 0x10, 0o666, 0); // write | create | append
```

Comment:
Since you're importing FileUtils, just use the constants [defined there](http://searchfox.org/mozilla-central/source/toolkit/modules/FileUtils.jsm#20) and get rid of the comment on this line. Or, better, why don't you just use [this](http://searchfox.org/mozilla-central/source/toolkit/modules/FileUtils.jsm#95) to open the file?

Reply:
Done